Automattic/wp-calypso
# Only fire tracks event when the reviews modal is visible

Related to https://github.com/Automattic/dotcom-forge/issues/5269

## Proposed Changes

* Add a condition to fire tracks event only when the modal is shown

## Testing Instructions



* Apply this patch and navigate to a plugin in an environment with the following flags enabled: `marketplace-add-review` and `marketplace-reviews-show`
* Open developer tools and activate logging for track events running `localStorage.debug='calypso:analytics'`. Make sure you have logging level set to `Verbose` in dev tools. Keep the Console tab open
* Go to a plugin details that you are able to review, e.g. `/plugin/woocommerce-bookings`
* Make sure that the `calypso_marketplace_reviews_modal_open` event is not fired yet
* Open the reviews modal (by either clicking the number of reviews in the plugin header, or the `Read all reviews` link in the reviews section)
* Make sure you can see a tracks event logged named `calypso_marketplace_reviews_modal_open` with the `productType` as `plugin` and the plugin slug as `slug`

![CleanShot 2024-01-22 at 17 03 54](https://github.com/Automattic/wp-calypso/assets/11555574/b27dcad8-387d-4901-ae6a-bf190df262d0)


## Pre-merge Checklist
